docs/design.md

```diff
@@ -0,0 +1,20 @@
+
```

Comment:
RJ thinks that document should be deleted

Reply:
done